messaging/coap/engine.c

```diff
@@ -521,6 +521,11 @@ coap_receive_method_payload(coap_receive_ctx_t *ctx, const char *href,
   }
 #endif /* OC_SECURITY */
 
+  if (!oc_main_initialized()) {
```

Comment:
This seems a bit late, can't we put this at the beginning of `coap_process_inbound_message` to skip the message receive at the start and avoid executing most of the code?

Reply:
We decided to keep it like this, maybe the check can be executed sooner, but that will require more extensive testing.